processor/src/errors.rs

```diff
@@ -50,11 +48,9 @@ impl Display for ExecutionError {
                 let hex = to_hex(Felt::elements_as_bytes(key))?;
                 write!(fmt, "Can't push values onto the advice stack: value for key {hex} not present in the advice map.")
             }
-            MerkleNodeIndex(depth, index) => write!(fmt, "The provided depth {depth} cannot be represented as Merkle index with the value {index}"),
+            MerkleNodeIndex { depth, value } => write!(fmt, "The provided depth {depth} cannot be represented as Merkle index with the value {value}"),
```

Comment:
I would update message to something like: "The specified index {index} is invalid for the specified depth {depth}"